File: Queues/CallWaitingSystem.java
/*
 File Name: CallWaitingSystem.java
 Program: Call Waiting System using Queue
*/

import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class CallWaitingSystem {

    static Queue<Integer> callQueue = new LinkedList<>();

    static void addCall(int callId) {
        callQueue.add(callId);
        System.out.println("Incoming call " + callId + " added to queue.");
    }

    static void handleCall() {
        if (callQueue.isEmpty()) {
            System.out.println("No calls waiting.");
        } else {
            int callId = callQueue.poll();
            System.out.println("Agent handling call " + callId + ".");
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int choice, callId;

        do {
            System.out.println("\n--- Call Waiting System ---");
            System.out.println("1. Add Incoming Call");
            System.out.println("2. Handle Call");
            System.out.println("3. Exit");
            System.out.print("Enter choice: ");
            choice = sc.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter Call ID: ");
                    callId = sc.nextInt();
                    addCall(callId);
                    break;
                case 2:
                    handleCall();
                    break;
                case 3:
                    System.out.println("Exiting system...");
                    break;
                default:
                    System.out.println("Invalid choice.");
            }
        } while (choice != 3);

        sc.close();
    }
}

File: Queues/call_waiting_system.py
# File Name: call_waiting_system.py
# Program: Call Waiting System using Queue

from collections import deque

call_queue = deque()

def add_call(call_id):
    call_queue.append(call_id)
    print(f"Incoming call {call_id} added to queue.")

def handle_call():
    if not call_queue:
        print("No calls waiting.")
    else:
        call_id = call_queue.popleft()
        print(f"Agent handling call {call_id}.")

while True:
    print("\n--- Call Waiting System ---")
    print("1. Add Incoming Call")
    print("2. Handle Call")
    print("3. Exit")

    choice = int(input("Enter your choice: "))

    if choice == 1:
        call_id = int(input("Enter Call ID: "))
        add_call(call_id)
    elif choice == 2:
        handle_call()
    elif choice == 3:
        print("Exiting system...")
        break
    else:
        print("Invalid choice.")
